src/vow.sol

```diff
@@ -101,40 +104,40 @@ contract Vow is LibNote {
     }
     // Pop from debt-queue
     function flog(uint era) external note {
-        require(add(era, wait) <= now);
+        require(add(era, wait) <= now, "Vow/wait-not-finished");
         Sin = sub(Sin, sin[era]);
         sin[era] = 0;
     }
 
     // Debt settlement
     function heal(uint rad) external note {
-        require(rad <= vat.dai(address(this)));
-        require(rad <= sub(sub(vat.sin(address(this)), Sin), Ash));
+        require(rad <= vat.dai(address(this)), "Vow/not-enough-dai");
+        require(rad <= sub(sub(vat.sin(address(this)), Sin), Ash), "Vow/not-enough-debt");
         vat.heal(rad);
     }
     function kiss(uint rad) external note {
-        require(rad <= Ash);
-        require(rad <= vat.dai(address(this)));
+        require(rad <= Ash, "Vow/not-enough-ash");
+        require(rad <= vat.dai(address(this)), "Vow/not-enough-dai");
         Ash = sub(Ash, rad);
         vat.heal(rad);
     }
 
     // Debt auction
     function flop() external note returns (uint id) {
-        require(sump <= sub(sub(vat.sin(address(this)), Sin), Ash));
-        require(vat.dai(address(this)) == 0);
+        require(sump <= sub(sub(vat.sin(address(this)), Sin), Ash), "Vow/debt-lower-sump");
```

Comment:
"Vow/insufficient-debt"?

Reply:
Or "Vow/not-enough-debt", to be consistent with "not-enough-surplus" below (actually I'd go w/"insufficient" for both, but it's not too big of a deal).